python/GafferUI/BoolPlugValueWidget.py

```diff
@@ -70,60 +77,69 @@ def setHighlighted( self, highlighted ) :
-					self.__boolWidget.setState( value )
+		self.__boolWidget.setErrored( errored )
 
-			self.__boolWidget.setErrored( value is None )
+		# Animation state
 
-			displayMode = Gaffer.Metadata.value( self.getPlug(), "boolPlugValueWidget:displayMode" )
-			if displayMode is not None :
-				displayMode = {
-					"switch" : self.__boolWidget.DisplayMode.Switch,
-					"checkBox" : self.__boolWidget.DisplayMode.CheckBox,
-					"tool" : self.__boolWidget.DisplayMode.Tool,
-				}.get( displayMode, self.__boolWidget.DisplayMode.CheckBox )
-				self.__boolWidget.setDisplayMode( displayMode )
+		animated = any( Gaffer.Animation.isAnimated( p ) for p in self.getPlugs() )
+		## \todo Perhaps this styling should be provided by the BoolWidget itself?
+		widgetAnimated = GafferUI._Variant.fromVariant( self.__boolWidget._qtWidget().property( "gafferAnimated" ) ) or False
+		if widgetAnimated != animated :
+			self.__boolWidget._qtWidget().setProperty( "gafferAnimated", GafferUI._Variant.toVariant( bool( animated ) ) )
+			self.__boolWidget._repolish()
 
-			## \todo Perhaps this styling should be provided by the BoolWidget itself?
-			animated = Gaffer.Animation.isAnimated( self.getPlug() )
-			widgetAnimated = GafferUI._Variant.fromVariant( self.__boolWidget._qtWidget().property( "gafferAnimated" ) ) or False
-			if widgetAnimated != animated :
-				self.__boolWidget._qtWidget().setProperty( "gafferAnimated", GafferUI._Variant.toVariant( bool( animated ) ) )
-				self.__boolWidget._repolish()
+		# Display mode and enabled status
+
+		displayMode = sole( Gaffer.Metadata.value( p, "boolPlugValueWidget:displayMode" ) for p in self.getPlugs() )
+		if displayMode is not None :
+			displayMode = {
+				"switch" : self.__boolWidget.DisplayMode.Switch,
+				"checkBox" : self.__boolWidget.DisplayMode.CheckBox,
+				"tool" : self.__boolWidget.DisplayMode.Tool,
+			}.get( displayMode, self.__boolWidget.DisplayMode.CheckBox )
+			self.__boolWidget.setDisplayMode( displayMode )
 
 		self.__boolWidget.setEnabled( self._editable( canEditAnimation = True ) )
 
 	def __stateChanged( self, widget ) :
 
-		self.__setPlugValue()
+		self.__setPlugValues()
 
 		return False
 
-	def __setPlugValue( self ) :
+	def __setPlugValues( self ) :
+
+		value = self.__boolWidget.getState()
+		assert( value != self.__boolWidget.State.Indeterminate ) # Should be set by us, not by user action
 
-		with Gaffer.UndoScope( self.getPlug().ancestor( Gaffer.ScriptNode ) ) :
+		with Gaffer.UndoScope( next( iter( self.getPlugs() ) ).ancestor( Gaffer.ScriptNode ) ) :
+			for plug in self.getPlugs() :
 
-			if Gaffer.Animation.isAnimated( self.getPlug() ) :
-				curve = Gaffer.Animation.acquire( self.getPlug() )
-				curve.addKey(
-					Gaffer.Animation.Key(
-						time = self.getContext().getTime(),
-						value = self.__boolWidget.getState(),
-						type = Gaffer.Animation.Type.Step
+				if Gaffer.Animation.isAnimated( plug ) :
+					curve = Gaffer.Animation.acquire( plug )
+					curve.addKey(
```

Comment:
Just a note, as this is historical, `NumericPlugValueWidget` only sets a key of the value is different.